core/domain/classifier_services.py

```diff
@@ -0,0 +1,322 @@
+# coding: utf-8
+#
+# Copyright 2015 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import operator
+
+import numpy
+
+
+class StringClassifier(object):
+    """Handles math-y internals for classifying strings.
+    https://en.wikipedia.org/wiki/Latent_Dirichlet_allocation"""
+
+    # Learning rates
+    _DEFAULT_ALPHA = 0.1
+    _DEFAULT_BETA = 0.001
+
+    def __init__(self):
+        """
+        Hieroglyph decoder:
+        b - boolean
+        c - count
+        i - id
+        l - label
+        w - word
+        d - doc
+
+        Internal (private) model representation:
+        _w_dc - word (id) given doc and count
```

Comment:
I think p for position makes sense, though 'd', 'w', and 'l' are all ids, just ids for their respective categories. Once the strings are passed in and run through _parse_examples, everything's an int. I'll make it clear in the doc.